### PR TITLE
Feature add auto mitre software code

### PR DESCRIPTION
**Pull request by cccs-gm**
Several feature updates as mentioned in the commits. 

Some work was done to fix issues caused by renaming some of the YaraFileProcessor function names. There are additional functions using the old function name that will be left for an as yet to be determined amount of time with deprecation messages.

Added the feature to generate MITRE ATT&CK software codes based on the value of malware metadata keys. This will allow for enrichment of yara rules with mitre_att software codes in many situations. The current functionality does not remove mitre_att software codes, so it is possible to have mitre_att software codes that do not have a matching malware name.

Added a couple of new warning checks.
  - if a info|exploit|technique|tool|malware key is not present a warning is raised. While not required it is a best practice to include the metadata with the additional information i.e. malware: "malware name"

  - Added a first iteration of warnings metadata keys that are similarly named to keys in the standard, this is to try and help catch things like hash1, hash2, or hash_256. This is a first pass and should be updated with more common items over time.

### CODE REVIEW

File: CCCS_YARA.yml [hunk 1]
```diff
@@ -206,6 +208,8 @@ mitre_att:
    unique: No
    optional: Yes
    validator: valid_mitre_att
+   argument:
+      softCodesFound:
```
Comment (cccs-jp) [inline CCCS_YARA.yml]: whats the meaning of softcodes?

Comment (cccs-jp) [reply CCCS_YARA.yml]: perhaps mitre_softwareid would be a better name

**Review by cccs-jp — CHANGES_REQUESTED** [review]

File: CCCS_YARA.yml (hunk 1 above)
Comment (cccs-gm) [reply CCCS_YARA.yml]: I will expand on the description as well. Those are being left there to indicate that these are fields that will be utilized by the validator to store found mitre_att software ids.

Comment (cccs-gm) [reply CCCS_YARA.yml]: Per the commit below I changed some of the names for more clarity:
  ifKeyGenMitreAtt -> if_key_gen_mitre_att
  malwareIdsFound  -> mitre_softwareid_gen
  softCodesFound   -> mitre_softwareid_found
I think those are a little clearer, I also added the support for tool: "toolname" metadata tags to auto generate mitre_att software codes if not present.

File: CCCS_YARA.yml [hunk 2]
```diff
@@ -182,15 +184,19 @@ category:
...
    format: 'uppercase'
    unique: No
    optional: Yes
    validator: valid_category_type
    argument:
       parent: category
-      ifKeyGenMitreAtt: malware
-      malwareIdsFound:
+      if_key_gen_mitre_att: ^tool$|^malware$
+      mitre_softwareid_gen: []
```
Comment (cccs-jp) [inline CCCS_YARA.yml]: should not be in the config file since it will never be populated unless at runtime

**Review by cccs-jp — APPROVED** [review]

**Comment by cccs-gm** [comment]
As discussed, merging.

**Pull request merged**